mlrun/runtimes/nuclio/function.py

```diff
@@ -544,6 +544,8 @@ def deploy(
         if tag:
             self.metadata.tag = tag
 
+        mlrun.utils.validate_tag_name(self.metadata.tag, "function.metadata.tag")
```

Comment:
please move to `pre_deploy_validation` - remember validations should be on server side